polls/views.py

```diff
@@ -25,12 +25,12 @@ def get(self, request):
         }
         return render(request, self.template_name, context)
 
-class TaskFlowView(LoginRequiredMixin, UserPassesTestMixin, View):
+class TaskFlowView(LoginRequiredMixin, View):
     template_name = "polls/task_flow.html"
 
     def get(self, request):
         can_continue, should_rest, _ = check_user_work_permission(request.user)
-        if should_rest:
+        if should_rest or request.user.is_locked:
```

Comment:
Uh this is wrong and badly designed.

a) If the user should rest, they don't go to the auth-flow. They get a page saying: "Thank you for your work come back in X minutes." or whatever I put in my spec. If you don't want to implement that yet, then you put a `#TODO: comment in the code`

b) You're doing an if check on `request.user.is_locked`. Those are going to be littered everywhere in the code and you're going to forget to add them somewhere. Why not just make a mixin or something that makes it really obvious which views / apps / whatever you can access if you are not frozen. And if you are frozen, you get a separate "Thank you for your work" page (or whatever I put in the spec, please read it and dont make me dig it up and repeat myself.) And if that's not ready then you put a `#TODO` there.

Reply:
a) That's already added to auth flow and I just didn't want to repeat it. I have implemented that already and I want to work on it.
b) You mean some decorator or mixin, if that's the case still we will have to add that custom decorator on all views. I will check for decorator or mixin.

Reply:
@Programmer-yar As discussed in telegram:

MAKE an issue to fix it, add a todo in the code, reply with a comment saying: I'll fix it in this issue.

Then we are DONE.

You can't go down endless rabbit holes. You need to break up the work into tiny bite sized pieces. Creating an issue to improve something later is a FAST way to be DONE.

Reply:
I will fix in this issue https://github.com/turian/audio-discrimination-crowdsource/issues/33